soc/nordic/common/soc_lrcconf.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc_lrcconf.h>
+#include <zephyr/kernel.h>
+
+static struct k_spinlock lock;
+static sys_slist_t poweron_main_list;
+static sys_slist_t poweron_active_list;
+
+void soc_lrcconf_poweron_request(sys_snode_t *node, nrf_lrcconf_power_domain_mask_t domain)
```

Comment:
any reason why simple reference counter is not used here? I only see that the difference is that you keep list of clients (but its not used anywhere) and allow to have non-symmetric requests (second request from the same client is not counted) but it can potentially cause some issues. Reference counter would be simpler and use less resources.

Reply:
At least right now there is non-symmetric requests so it couldn't be dropped

Reply:
I think the code is symmetric now, maybe this would benefit from an onoff manager style thing for reference counting instead of manually tracking the references?